obspy/io/nordic/core.py

```diff
@@ -0,0 +1,1379 @@
+    '12.3'
+    """
+    string = str(number)
+    if (isinstance(number, float) and np.isnan(number)) or number == 999:
+        string = ' '
+    elif not rounded and \
+            (isinstance(number, float) or isinstance(number, int)):
+        if number < 100000:
+            string = str(number)
+        else:
+            exponant = int('{0:.2E}'.format(number).split('E+')[-1]) - 1
+            divisor = 10 ** exponant
+            string = '{0:.1f}'.format(number / divisor) + 'e' + str(exponant)
+    elif rounded == 2 and \
+            (isinstance(number, float) or isinstance(number, int)):
+        string = '{0:.2f}'.format(number)
+    elif rounded == 1 and \
+            (isinstance(number, float) or isinstance(number, int)):
+        string = '{0:.1f}'.format(number)
+    else:
+        return str(number)
+    return str(string)
+
+
+def _evmagtonor(mag_type):
+    """
+    Switch from obspy event magnitude types to seisan syntax.
+
+    >>> _evmagtonor('mB')
+    'b'
+    >>> _evmagtonor('M')
+    'W'
+    >>> _evmagtonor('bob')
+    ''
+    """
+    mag_type = str(mag_type)
+    if mag_type in ['ML', 'MLv']:
+        # MLv is local magnitude on vertical component
+        mag = 'L'
+    elif mag_type == 'mB':
+        mag = 'b'
+    elif mag_type == 'Ms':
+        mag = 's'
+    elif mag_type == 'MS':
+        mag = 'S'
+    elif mag_type in ['MW', 'Mw']:
+        mag = 'W'
+    elif mag_type == 'MbLg':
+        mag = 'G'
+    elif mag_type in ['Mc', 'MC']:
+        mag = 'C'
+    elif mag_type == 'M':
+        mag = 'W'  # Convert generic magnitude to moment magnitude
+        msg = ('Converting generic magnitude to being stored as moment mag')
+        warnings.warn(msg)
+    else:
+        warnings.warn(mag_type + ' is not convertible')
+        return str('')
+    return str(mag)
+
+
+def _nortoevmag(mag_type):
+    """
+    Switch from nordic type magnitude notation to obspy event magnitudes.
+
+    >>> _nortoevmag('b')
+    'mB'
+    >>> _nortoevmag('bob')
+    ''
+    """
+    mag_type = str(mag_type)
+    if mag_type == 'L':
+        mag = 'ML'
+    elif mag_type == 'b':
+        mag = 'mB'
+    elif mag_type == 's':
+        mag = 'Ms'
+    elif mag_type == 'S':
+        mag = 'MS'
+    elif mag_type == 'W':
+        mag = 'MW'
+    elif mag_type == 'G':
+        mag = 'MbLg'
+    elif mag_type == 'C':
+        mag = 'Mc'
+    else:
+        warnings.warn(mag_type + ' is not convertible')
+        return str('')
+    return str(mag)
+
+
+def readheader(sfile):
+    """
+    Read header information from a seisan nordic format S-file.
+    Returns an obspy.core.event.Catalog type: note this changed for version \
+    0.1.0 from the inbuilt class types.
+
+    :type sfile: str
+    :param sfile: Path to the s-file
+
+    :returns: :class: obspy.core.event.Event
+    """
+    with open(sfile, 'r') as f:
+        header = _readheader(f=f)
+    return header
+
+
+def _readheader(f):
+    """Internal header reader.
```

Comment:
Please use

```python
def function(...):
    """ (NEWLINE!)
    docstring
    """
    ...
```

everywhere to stay consistent with the rest of ObsPy. You mostly did this everywhere but there are a couple places where you did not.